lib/router/router.ts

```diff
@@ -120,9 +120,19 @@ export default abstract class Router<T extends Route> {
       // perform a URL update at the end. This gives
       // the user the ability to set the url update
       // method (default is replaceState).
-      let newTransition = new InternalTransition(this, undefined, undefined);
+      let newTransition = new InternalTransition(
+        this,
+        undefined,
+        undefined,
+        undefined,
+        this.activeTransition
+      );
```

Comment:
I'm attempting to fix a semi-related bug (https://github.com/emberjs/ember.js/issues/18577) in #307.

```diff
- let newTransition = new InternalTransition(this, undefined, undefined);
+ let newTransition = new InternalTransition(this, undefined, newState);
```